From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: John Kleinschmidt <jkleinsc@electronjs.org>
Date: Tue, 27 Apr 2021 11:40:59 -0400
Subject: debug websql error

this is a temporary patch to debug the websql error on WOA.

diff --git a/third_party/blink/renderer/modules/webdatabase/sqlite/sqlite_database.cc b/third_party/blink/renderer/modules/webdatabase/sqlite/sqlite_database.cc
index b35b16647701f0e2a350cf6538664654553704d8..39630ffd52ef483d3b52d0c0eb3e76e7111562d5 100644
--- a/third_party/blink/renderer/modules/webdatabase/sqlite/sqlite_database.cc
+++ b/third_party/blink/renderer/modules/webdatabase/sqlite/sqlite_database.cc
@@ -33,6 +33,7 @@
 #include "third_party/blink/renderer/modules/webdatabase/sqlite/sql_log.h"
 #include "third_party/blink/renderer/modules/webdatabase/sqlite/sqlite_statement.h"
 #include "third_party/sqlite/sqlite3.h"
+#include "base/debug/stack_trace.h"
 
 namespace blink {
 
@@ -133,8 +134,10 @@ bool SQLiteDatabase::Open(const String& filename) {
 
   opening_thread_ = CurrentThread();
 
-  if (!SQLiteStatement(*this, "PRAGMA locking_mode = NORMAL;").ExecuteCommand())
+  if (!SQLiteStatement(*this, "PRAGMA locking_mode = NORMAL;").ExecuteCommand()) {
     DLOG(ERROR) << "SQLite database could not set locking_mode to normal";
+    base::debug::StackTrace().Print();
+  }
 
   if (!SQLiteStatement(*this, "PRAGMA temp_store = MEMORY;").ExecuteCommand())
     DLOG(ERROR) << "SQLite database could not set temp_store to memory";
